src/lib.rs

```diff
@@ -1,7 +1,9 @@
+#![allow(incomplete_features)] // Used for generic_const_exprs
 #![cfg_attr(not(feature = "std"), no_std)]
 #![feature(cfg_version)]
 #![feature(generic_associated_types)] // For mutex
 #![feature(const_fn_fn_ptr_basics)] // Embassy timer
```

Comment:
I'm digressing, but: I'll remove this one myself too, as we no longer support Embassy timer (turns out it cannot be supported on executors other than Emassy's own one, and Embassy executor being non-thread based is not a good fit for the `esp-idf-*` crates, unlike `smol`).